src/register/index.js

```diff
@@ -2,19 +2,27 @@ import check from '../check/index.js';
 import { code, message } from '../consts/index.js';
 
 /**
- * Register "checkAndThrow" function to the `window.load` event.
+ * Call "checkAndThrow" only when document's readyState is complete.
+ * @param {function} callback
+ * @returns {undefined}
  */
-export default function register() {
+export default function register(callback) {
     document.readyState === 'complete'
-        ? checkAndThrow()
-        : window.addEventListener('load', checkAndThrow);
+        ? checkAndThrow(callback)
+        : document.addEventListener(
+            'readystatechange',
+            () => register(callback),
+            { once: true }
+        );
 }
 
 /**
  * Belated execution of the check and error throwing
  * Throws the error globally (potentially).
+ * @param {function} callback
+ * @returns {undefined}
  */
-function checkAndThrow() {
+function checkAndThrow(callback) {
```

Comment:
Now after the throw is conditional, maybe just `check` would do here?

Reply:
check is already in the API. How about "checkAndErr"?

Reply:
Oh, didn't catch that 👍 WDYT on just checkErrors

Reply:
I meant err as a verb, actually. To err is to make an error.
In a way that the page violates an accepted standard, it errs.

How about "createError"?